python/ray/autoscaler/kubernetes/config.py

```diff
@@ -151,3 +156,38 @@ def _configure_autoscaler_role_binding(namespace, provider_config):
     logger.info(log_prefix + not_found_msg(binding_field, name))
     auth_api().create_namespaced_role_binding(namespace, binding)
     logger.info(log_prefix + created_msg(binding_field, name))
+
+
+def _configure_services(namespace, provider_config):
+    service_field = "services"
+    if service_field not in provider_config:
+        logger.info(log_prefix + not_provided_msg(service_field))
+        return
+
+    services = provider_config[service_field]
+    for service in services:
+        if "namespace" not in service["metadata"]:
+            service["metadata"]["namespace"] = namespace
+        elif service["metadata"]["namespace"] != namespace:
+            raise InvalidNamespaceError(service_field, namespace)
+
+        name = service["metadata"]["name"]
+        field_selector = "metadata.name={}".format(name)
+        services = core_api().list_namespaced_service(
+            namespace, field_selector=field_selector).items
+        if len(services) > 0:
+            assert len(services) == 1
+            existing_service = services[0]
+            if service == existing_service:
+                logger.info(log_prefix +
+                            using_existing_msg(service_field, name))
+                return
+            else:
+                logger.info(log_prefix +
+                            deleting_existing_msg(service_field, name))
+                core_api().delete_namespaced_service(name, namespace)
```

Comment:
Is it significantly harder to patch the config for the existing service?

I'm worried about the trapdoor nature of delete releasing resources (I'm thinking cluster IPs specifically).

Reply:
Nope, updated!